### ISSUE TITLE
read_owned and write_owned

### ISSUE THREAD

**Issue by sinhpham** [yoshuawuyts/fd-lock#29]
My use case: keeping the lock guard tied to an object

```
struct MyStruct {
    // .... data
    lock: OwnedRwLockWriteGuard
}
```

It's similar to tokio's `RwLock` API: https://docs.rs/tokio/latest/tokio/sync/struct.RwLock.html#method.write_owned

**Comment by sinhpham** [comment]
@yoshuawuyts @sunfishcode I can work on a PR if this is a feature `fd-lock` wants to provide.

**Comment by sunfishcode** [comment]
Yes, I believe it makes sense for fd-lock to provide this, and am able to review patches for it.

**Comment by dpc** [comment]
I just came to report that this crate has a weird API.


I keep trying to use `fd-lock` in my projects, because I'm a forgetful dummy, and it appears on top of [crates.io search](https://crates.io/search?q=fs%20lock&sort=recent-downloads), and then I keep hitting these artificail limitations, where all I really wanted was something to call the unlock on drop on the file descriptor.

The owned guards are a part of it, but eg `RwLock::write` taking `&mut self` is another one. An FS lock is NOT a Mutex involved in a, what's the point of `&mut`?

There doesn't need to be "read" vs "guard" distinction. It's not a `std::sync::RwLock`. Creating two just makes it neccessary to create `enum Guard { Read(...), Write(...) }` if you need to store them. While in reality both Unix and Windows will call the same thing under the hood, and any different
